Source/dom/components/Library.js

```diff
@@ -0,0 +1,141 @@
+import { WrappedObject, DefinedPropertiesKey } from '../WrappedObject'
+import { Document } from './Document'
+import { toArray, getURLFromPath } from '../utils'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { wrapObject } from '../wrapNativeObject'
+import { ShareableObject } from './ShareableObject'
+
+const AddStatus = {
+  0: 'ok',
+  1: 'the library has already been added',
+  2: 'the document is not in the new JSON format',
+  3: 'there was a problem reading the asset library file',
+}
+
+/**
+ * A Sketch Library.
+ */
+export class Library extends WrappedObject {
+  constructor(library = {}) {
+    if (!library.sketchObject) {
+      throw new Error('Cannot create a new Library directly')
+    }
+
+    super(library)
+  }
+
+  static getLibraries() {
+    const libraryController = AppController.sharedInstance().librariesController()
+    return toArray(libraryController.libraries()).map(
+      Library.fromNative.bind(Library)
+    )
+  }
+
+  static createLibraryFromDocument(document, path) {
+    if (typeof document === 'string' && !path) {
+      /* eslint-disable no-param-reassign */
+      path = document
+      document = undefined
+      /* eslint-enable */
+    }
+
+    const libUrl = getURLFromPath(path)
+
+    if (document) {
+      const wrappedDocument = wrapObject(document)
+      wrappedDocument.save(path)
+    }
+
+    const libraryController = AppController.sharedInstance().librariesController()
+    const status = libraryController.addAssetLibraryAtURL(libUrl)
+
+    if (status !== 0) {
+      throw new Error(`Error while adding the library: ${AddStatus[status]}.`)
+    }
+
+    // refresh the UI
+    libraryController.notifyLibraryChange(null)
+
+    const lib = toArray(libraryController.userLibraries()).find(l =>
+      l.locationOnDisk().isEqual(libUrl)
+    )
```

Comment:
This isn't wrong, but the newly added library will alway be first in the list so you could just do `firstObject` instead.

Reply:
it is wrong if another library was added at the same path and then deleted? (just trying to understand how this is wrong but will use `firstObject` even tho that seems fragile...)

Reply:
If another library had been added at that path but NOT removed from the library controller then `addAssetLibraryAtURL` would fail. In this case when the new document appears at the same location the library controller would find it and the existing library would attach to it.

I'm not saying what you're doing is wrong BTW. I'm just pointing out that it could be simplified as `addAssetLibraryAtURL` will always add to the start of the list if it returns success. I'm happy for you to keep it this way if you're worried that it's fragile though.